cdap-security/src/main/java/io/cdap/cdap/security/authorization/DefaultAccessEnforcer.java

```diff
@@ -341,6 +365,41 @@ private static void addTagsForEntityId(Map<String, String> tags, EntityId entity
     }
   }
 
+  private void incrementCheckMetricExternal(MetricsContext metricsContext,
+                                            AuthorizationResponse authorizationResponse) {
```

Comment:
Why do you need two separate calls? Why the enforce can't return `UNAUTHORIZED`?

Reply:
`UNAUTHORIZED` will fall under default : Failure.

Reply:
But why? Why do you need to differenciate external and internal and use different metric for UNAUTHORIZED?

Reply:
In `Constants.Metrics.Authorization` we have metrics that are mostly `INTERNAL_`.  or `EXTENSION_` 

In previous code we were putting the metric right after the call. i.e. `INTERNAL_` for `internalAccessEnforcer` calls and `EXTENSION_`  `accessControllerInstantiator.get()` calls

So I thought of having 2 methods for `INTERNAL_`.  or `EXTENSION_`  

----
for `use different metric for UNAUTHORIZED?` 
In `EXTENSION_`  there is `success`, `failure`, `bypass`  
For i thought UNAUTHORIZED will increment `failure` ( also also rest of the cases apart from AUTHORIZED and NOT_REQUIRED)

Let me know how to handle this.

Reply:
INTERNAL_ metric applies only to internalAccessEnforcer, nothing else. Everything else should be EXTENSION and this is what we care most about and should have extended data. Note that it's EXTENSION, not external. In other words, it's about intraservice authorization (internal) vs user authorization (extension)